content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java

```diff
@@ -659,23 +664,32 @@ public void processUnsubmitted(Calendar cal) {
 				errors++;
 				continue;
 			}
-			// Get filename of submission
-			String fileName = resource.getProperties().getProperty(ResourceProperties.PROP_DISPLAY_NAME);
-			if (StringUtils.isEmpty(fileName)) {
-				// Set default file name:
-				fileName = "submission_" + item.getUserId() + "_" + item.getSiteId();
-				log.info("Using Default Filename " + fileName);
-			}
 			// EXTERNAL ID DOES NOT EXIST, CREATE SUBMISSION AND UPLOAD CONTENTS TO TCA
 			// (STAGE 1)
 			if (StringUtils.isEmpty(item.getExternalId())) {
+				// Get filename of submission
+				String fileName = resource.getProperties().getProperty(ResourceProperties.PROP_DISPLAY_NAME);
+				// If file name is missing extension add to fulfill TCA header requirement
+				if (FilenameUtils.getExtension(fileName).isEmpty()) {
+					// Add .html for in line submissions
+					if (INLINE_SUBMISSION.equals(fileName)) {
+						fileName += HTML_EXTENSION;
```

Comment:
why is docx the default? Is there a more generic option?

Reply:
perhaps .txt would be better here?

Reply:
let's leave this for now, we may remove the default completely if ends up not being required

Reply:
I talked w/Chris... do not set an extension if you don't know it. In the "inline submission" case, we know it is "html", so keep that... remove the default docx completely

Reply:
okay, what about for default filename?

Reply:
Leave a default filename, just w/o extension